src/content/1.7/basics/keeping_up-to-date/migration.md

```diff
@@ -0,0 +1,678 @@
+```
+
+If you want to know in which folder you can export the files, run the SQL request:
+
+```sql
+mysql> SHOW VARIABLES LIKE "secure_file_priv";
+```
+
+### Updating the data
+
+#### Get changes between source and destination databases
+Now you have retrieved your data from the production shop, you have to adapt it to the new structure before inserting it on your new database.
+
+An efficient way to get the changes is getting a database structure diff between the 2 running shops. MySQL provides another tool for this called [mysqldiff](https://dev.mysql.com/doc/mysql-utilities/1.6/en/mysqldiff.html).
+
+Note this utility may be not immediately available on your environment, even if you have the mysql client installed. For instance on Ubuntu, an additional package needs to be installed:
+
+```bash
+apt install mysql-utilities
+```
+
+This tool can be run anywhere, as long as you can access both servers & databases. The basic use of this command requires the following parameters:
+
+```txt
+Usage: mysqldiff --server1=user:pass@host:port:socket --server2=user:pass@host:port:socket db3:db4
+```
+
+You need to provide the credentials to your MySQL servers. Ifthe source and the destination databases are on the same server, the `--server2` parameter can be omitted. `db3` and `db4` are respectively the source and the destination databases of your data.
+
+Getting the structure diff will require additional options, as provided in this example:
+```bash
+mysqldiff --server1=root:xxxx@127.0.0.1:3307 --changes-for=server1 --skip-table-options --force prestashop16:prestashop
+```
+
+* All our databases are on the same server, that’s why only `--server1` is provided,
+* We want to know what are the changes to apply on the 1.6 version to fit the 1.7 structure, so we need `--changes-for=server1`
+* We only need structure changes. `--skip-table-options` removes diff about autoincrement etc.
+* By default `mysqldiff` will only tell you the databases are not the same and will return immediately. Providing `--force` will make the tool print out details about the differences found.
+
+A example of output is available in https://gist.github.com/Quetzacoalt91/9cef61ebbf12c139f44d27035d3cb1fd. The changes are extracted from a diff structure between a PrestaShop 1.6.1.18 database and another one under PrestaShop 1.7.3.2.
+
+It appears the structure from PrestaShop has not drastically changed from 1.6 versions, although some columns type has evolved. Preparing your data for PrestaShop 1.7 should not need too many changes on your CSV files.
+
+### Applying structure change
+This step has to be reproduced as many times as you have tables waiting to be migrated. The import step will be easier if you make sure the data you transfer fits the destination structure. This chapter will help to identify the changes to apply depending on the diff created during the previous step.
+
+A few notes:
```

Comment:
Remember to append an empty line before the list, or else it won't be rendered properly.